### DISCUSSION TITLE
Plans for iOS 15?

### DISCUSSION THREAD

**Discussion by PippoApps** [airsdk/Adobe-Runtime-Support#899] [General]
Hello there, 
as you know, Apple is about to launch iOS 15.
Is AIR going to be ready before the new OS will be shipped on all devices?

Cheers
Filippo

**Comment by marchbold** [comment]
Definitely keen to hear the plans for this as well.

**Comment by leossmith** [comment]
Also interested hearing about the plans for it.

**Comment by ajwfrost** [comment]
We've just been analyzing this - there's some work to do here, but we'll aim to get this working ahead of time once the final releases of Xcode 13 / iPhoneOS 15 are out...  doesn't look like any major concerns that we can see though.

**Reply by PippoApps** [reply]
Thank you Andrew.
Sounds perfect.

**Comment by d-angelovski** [comment]
I tried the app on iOS 15, and it won't run at all (see attached image). On previous versions of iOS was just showing the message that "This app will not work with future versions of iOS. The developer of this app needs to update it to improve its compatibility."

I thought this will be solved with building the ipa from latest Mac OS (big sur) and latest Air (533), but yesterdays build did not work either.
![IMG_1262](https://user-images.githubusercontent.com/23104582/124273746-15d04b80-db49-11eb-9a19-11770c6fc548.PNG)

Is this issue connected, or we have some ANE not compatible, or something else we are missing in the build process?

**Reply by d-angelovski** [reply]
Just to fill in on details, I did an empty project without ANE's and did try to rebuild app and made new .ipa. They also show the same message as above, so definitely its something wrong on IOS 15 and apps will not work.

**Reply by ajwfrost** [reply]
@d-angelovski  thanks for checking .. can I clarify, it did install though? So it installed okay but when you ran it, it came up with this display?

**Reply by d-angelovski** [reply]
> @d-angelovski thanks for checking .. can I clarify, it did install though? So it installed okay but when you ran it, it came up with this display?

It installed, but immediately after installation, the screen pops up. If you select Keep, then again on next app start (clicking the icon), it says the same message. Even with just a blank project without any Ane or code. But it never lets you in the app, so its going to be big issue. Until now (iOS 14.x), there was popup saying that "In future versions it will not work" but it was letting you in the app. I thought that it will be fixed with air 33  - 533 (however, that has some other issues that probably is for another time).

**Reply by FliplineStudios** [reply]
@d-angelovski That's strange, we also saw that same popup in iOS 14.x with older AIR SDKs, but once we switched to .476 with the newer codesigning we stopped getting that popup in iOS 14, so it's odd you were still seeing it in iOS 14 before this....  what version of AIR were you using previous to .533?  It also might be worth double-checking the IPA to make sure it's really being built with .533 and isn't accidentally using an older version still (the quickest way is probably to unzip the IPA and check the Info.plist for the CTAirSdkVersion key).

**Reply by d-angelovski** [reply]
> @d-angelovski ...(the quickest way is probably to unzip the IPA and check the Info.plist for the CTAirSdkVersion key).

I have just did that, its latest version:
<key>CTAirSdkVersion</key>
<string>33.1.1.533</string>

I will try with earlier also, but I am pretty sure we were experimenting with it also before.

**Reply by ajwfrost** [reply]
If you've unzipped, could you please also run from a command-line:
```
codesign -dvvvv Payload/appname.app
```
so we can check what format has gone in there?

thanks

**Reply by d-angelovski** [reply]
Here it goes:

Identifier=com.servsuitemobile....
Format=app bundle with Mach-O thin (arm64)
CodeDirectory v=20400 size=483111 flags=0x0(none) hashes=15088+5 location=embedded
VersionPlatform=2
VersionMin=851968
VersionSDK=918784
Hash type=sha256 size=32
CandidateCDHash sha1=......
CandidateCDHash sha256=.....
Hash choices=sha1,sha256
Executable Segment base=0
Executable Segment limit=59228160
Executable Segment flags=0x1
Page size=4096
CDHash=.....
Signature size=4785
Authority=iPhone Distribution: Service Pro.net Inc.
Authority=Apple Worldwide Developer Relations Certification Authority
Authority=Apple Root CA
Signed Time=Jul 2, 2021 at 10:59:09 PM
Info.plist entries=61
TeamIdentifier=.....
Sealed Resources version=2 rules=10 files=615
Internal requirements count=1 size=196

Placing ... for (maybe) not so public data.

**Reply by ajwfrost** [reply]
Thanks... so version 20400 for the code directory information is good, and version 918784 is 0xE0500 so 14.5 which is correct based on what we had built the SDK from.. I can't see anything wrong there.

Do you have any IPAs that do work on it, that you could check to compare these details against?

We've got AIR building on Xcode 13/iPhoneOS SDK 15, beta versions, so will also check whether we can reproduce the above and whether the updated build settings make any difference..

thanks

**Reply by avelle** [reply]
I have the same problem here,need help.

**Reply by ajwfrost** [reply]
Hi @d-angelovski , @PippoApps 
It turns out there was a format change in the code signature, although still within the 20400 code directory version.. where it had said "xxx + 5" hashes, we needed to update it to "xxx + 7" and to include an extra hash of the entitlements, this time DER-encoded ...
So: we have an updated version of ADT for now just to get something that you can use, see the below link:
https://transfer.harman.com/message/Hn0uKXIqQQHARz1dRhERvH

and for more details: https://github.com/airsdk/Adobe-Runtime-Support/discussions/1063#discussioncomment-1236287

Please let us know of any issues!
thanks

**Reply by d-angelovski** [reply]
Hi @ajwfrost ,

I had tried the latest air 33 - 575 (without the ADT.jar provided in the link), and with a cfg option: UseNativeCodesign=true, and that worked on the beta IOS 15.

When you gave the ADT.jar, I tried with that, on top of 575, and with UseNativeCodesign=true commented out, it also worked OK.

So I think I can confirm that it works.

Its good to have that option included in the CFG file (commented out, like the others), since I found it trough out the threads here.

Thanks.

**Comment by SUSDEV** [comment]
@ajwfrost I just upgraded my iPad Pro 2019 to iOS 15 Beta 7. My app (made with 33.1.1.216) is working fine as-is, downloaded from the App Store. I haven't tested all the functions. Has anyone else tried Beta 7? Of course at some point I want to update the app, but the urgency is less at this point.
